tests/tests.py

```diff
@@ -236,11 +235,72 @@ def test_syllabify_text(self):
             [["ecce"], ["enim"], ["ex"], ["hoc"], ["be-", "a-", "tam"]],
         ]
         with self.subTest("Presyllabified Text"):
+            syllabified_text = syllabify_text(presyllabified_text, text_presyllabified=True)
+            syllabified_text_list = [section.section for section in syllabified_text]
             self.assertEqual(
-                syllabify_text(presyllabified_text, text_presyllabified=True),
+                syllabified_text_list,
                 expected_result,
             )
 
+class TestVolpianoSyllabification(unittest.TestCase):
+    """
+    Tests functions for syllabifying volpiano in volpiano_syllabification.py.
+    """
+    def test_preprocess_volpiano(self):
+        with self.subTest("Remove starting material"):
+            standard_volpiano = "1---g---h---3"
+            volpiano_with_extra_starting_matter = "tf-g-1---g-1--h---3"
+            expected = ("g---h---", "3")
+            self.assertEqual(preprocess_volpiano(standard_volpiano), expected)
+            self.assertEqual(preprocess_volpiano(volpiano_with_extra_starting_matter), expected)
+
+    def test_syllabify_volpiano(self):
+        volpiano = (
+            # Section divided by barline "3" + standard spacing
+            "a-b--c---d---e---3---"
+            # Section divided by barline "4" + standard spacing
+            "a-b--c---d---e---4---"
+            # Section divided by barline "3" + non-standard spacing
+            "a-b--c---d---e-3--"
+            # Section divided by barline "4" + break encoded
+            "a-b--c---d---e---47---"
+            # Missing music section
+            "a-b--c---6------6---"
+            # Missing music section with break encoded
+            "a-b--c---6------6---77"
+        )
+        expected = [
+            [["a-b--","c---"], ["d---"], ["e---"]], [["3---"]],
+            [["a-b--","c---"], ["d---"], ["e---"]], [["4---"]],
+            [["a-b--","c---"], ["d---"], ["e-"]], [["3--"]],
```

Comment:
This is correct. My general approach was that where we can infer with a high degree of confidence what was meant, we should infer it. Other functions along the way will take care of alignment because, in part, the spacing of alignment might also depend on the length of text.

Now, as per conversations we've had as recently as last lab meeting, eventually we may want to offer "feedback" at the point of entry on Cantus DB; we might want to raise an "error" when someone enters something like `e-3---` and encourage them to change it to the correct encoding. Or, once the volpiano starts coming from the OMR process, we might want to raise a string thus encoded for a human to check. This is an enhancement that is the subject of #5. But if/when this is eventually implemented, I imagine we'll want it to be something that can be "turned off" -- "errors" like this should be raised when volpiano is created, but if there's "bad" volpiano somewhere already in CantusDB, we want to do our best to render something.